File: go/subscription.go
// this file is @generated
package meteroid

import "time"

type Subscription struct {
	// When the subscription was activated (first payment or activation condition met)
	// RFC 3339 timestamp.
	ActivatedAt *time.Time `json:"activated_at,omitempty"`

	// If false, invoices will stay in Draft until manually reviewed and finalized. Default to true.
	AutoAdvanceInvoices bool `json:"auto_advance_invoices"`

	BillingDayAnchor int32 `json:"billing_day_anchor"`

	// When billing started (after any trial period)
	BillingStartDate *string `json:"billing_start_date,omitempty"`

	// Automatically try to charge the customer's configured payment method on finalize.
	ChargeAutomatically bool `json:"charge_automatically"`

	// When the subscription was created
	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Currency Currency `json:"currency"`

	// Current billing period end date
	CurrentPeriodEnd *string `json:"current_period_end,omitempty"`

	// Current billing period start date
	CurrentPeriodStart string `json:"current_period_start"`

	// User-defined custom property values, keyed by definition `key`.
	CustomProperties RequiredMap[any] `json:"custom_properties"`

	CustomerAlias *string `json:"customer_alias,omitempty"`

	CustomerId CustomerId `json:"customer_id"`

	CustomerName string `json:"customer_name"`

	// When the subscription ends (if set)
	EndDate *string `json:"end_date,omitempty"`

	Id SubscriptionId `json:"id"`

	// Default memo for invoices
	InvoiceMemo *string `json:"invoice_memo,omitempty"`

	// Monthly recurring revenue in cents
	MrrCents int64 `json:"mrr_cents"`

	// Payment terms in days (0 = due on issue)
	NetTerms int32 `json:"net_terms"`

	PaymentMethodsConfig *PaymentMethodsConfig `json:"payment_methods_config,omitempty"`

	// Billing period (monthly, annual, etc.)
	Period BillingPeriodEnum `json:"period"`

	PlanDescription *string `json:"plan_description,omitempty"`

	PlanId PlanId `json:"plan_id"`

	PlanName string `json:"plan_name"`

	PlanVersion int32 `json:"plan_version"`

	PlanVersionId PlanVersionId `json:"plan_version_id"`

	PurchaseOrder *string `json:"purchase_order,omitempty"`

	// When the subscription contract starts (benefits apply from this date)
	StartDate string `json:"start_date"`

	Status SubscriptionStatusEnum `json:"status"`

	// Trial duration in days
	TrialDuration *int32 `json:"trial_duration,omitempty"`
}

File: python/meteroid/models/subscription.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .billing_period_enum import BillingPeriodEnum
from .currency import Currency
from .customer_id import CustomerId
from .payment_methods_config import PaymentMethodsConfig
from .plan_id import PlanId
from .plan_version_id import PlanVersionId
from .subscription_id import SubscriptionId
from .subscription_status_enum import SubscriptionStatusEnum


@dataclasses.dataclass
class Subscription(BaseModel):
    auto_advance_invoices: bool
    """If false, invoices will stay in Draft until manually reviewed and finalized. Default to true."""

    billing_day_anchor: int

    charge_automatically: bool
    """Automatically try to charge the customer's configured payment method on finalize."""

    created_at: datetime
    """When the subscription was created"""

    currency: Currency

    current_period_start: str
    """Current billing period start date"""

    custom_properties: t.Dict[str, t.Any]
    """User-defined custom property values, keyed by definition `key`."""

    customer_id: CustomerId

    customer_name: str

    id: SubscriptionId

    mrr_cents: int
    """Monthly recurring revenue in cents"""

    net_terms: int
    """Payment terms in days (0 = due on issue)"""

    period: BillingPeriodEnum
    """Billing period (monthly, annual, etc.)"""

    plan_id: PlanId

    plan_name: str

    plan_version: int

    plan_version_id: PlanVersionId

    start_date: str
    """When the subscription contract starts (benefits apply from this date)"""

    status: SubscriptionStatusEnum

    activated_at: t.Optional[datetime] = None
    """When the subscription was activated (first payment or activation condition met)"""

    billing_start_date: t.Optional[str] = None
    """When billing started (after any trial period)"""

    current_period_end: t.Optional[str] = None
    """Current billing period end date"""

    customer_alias: t.Optional[str] = None

    end_date: t.Optional[str] = None
    """When the subscription ends (if set)"""

    invoice_memo: t.Optional[str] = None
    """Default memo for invoices"""

    payment_methods_config: t.Optional[PaymentMethodsConfig] = None

    plan_description: t.Optional[str] = None

    purchase_order: t.Optional[str] = None

    trial_duration: t.Optional[int] = None
    """Trial duration in days"""
